### PR TITLE
SC-13

### PR DESCRIPTION
**Pull request by mahir0837**

### CODE REVIEW

File: src/com/company/balance/Balance.java [hunk 1]
```diff
@@ -13,7 +13,7 @@ public Balance(UUID customerId, Double balance) {
         this.balance = balance;
     }
 
-    public abstract Double addBalance(Double additionalBalance);
```
Comment (ihccik) [inline src/com/company/balance/Balance.java]: Please revert this change not to brake the main flow

File: src/com/company/balance/CustomerBalance.java [hunk 2]
```diff
@@ -9,8 +10,13 @@ public CustomerBalance(UUID customerId, Double balance) {
...
     @Override
-    public Double addBalance(Double additionalBalance) {
-        setBalance(getBalance() + additionalBalance);
+    public Double addBalance(int selection) {
+        if (selection==1){
+            System.out.println("How much you would like to add?");
+            double additionalAmount = new Scanner(System.in).nextDouble();
+            setBalance(getBalance() + additionalAmount);
+            System.out.println("New customer balance: " + getBalance());
+        }
```
Comment (ihccik) [inline src/com/company/balance/CustomerBalance.java]: This part is breaking "Single Responsibility" rule of SOLID.
Because, printing something on the console and getting user input is not this class' responsibility.
Please revert this change not to brake the main flow

File: src/com/company/balance/GiftCardBalance.java [hunk 3]
```diff
@@ -12,9 +13,15 @@ public GiftCardBalance(UUID customerId, Double balance) {
...
-        double promotionAmount = additionalBalance * 10 / 100;
-        setBalance(getBalance() + additionalBalance + promotionAmount);
+    public Double addBalance(int selection) {
+        if (selection==2){
+            System.out.println("How much you would like to add?");
+            double additionalAmount = new Scanner(System.in).nextDouble();
+            double promotionAmount = additionalAmount * 10 / 100;
+            setBalance(getBalance() + additionalAmount + promotionAmount);
+            System.out.println("New giftCard balance: " + getBalance());
+        }
```
Comment (ihccik) [inline src/com/company/balance/GiftCardBalance.java]: This part is breaking "Single Responsibility" rule of SOLID.
Because, printing something on the console and getting user input is not this class' responsibility.
Please revert this change not to brake the main flow

**Review by ihccik — CHANGES_REQUESTED** [review]
You are doing in the hard way.
My advice is to revert all changes and just create a method in the Main class, then move all the code of the "add balance" case to this new method, just call this method within the case.

File: src/com/company/balance/Balance.java (hunk 1 above)
Comment (mahir0837) [reply src/com/company/balance/Balance.java]: I`ve changed my solution base on the your advice. Can you please check it and review again?

File: src/com/company/balance/CustomerBalance.java (hunk 2 above)
Comment (mahir0837) [reply src/com/company/balance/CustomerBalance.java]: I have reverted and changed my solution base on the "Single Responsibility" rule of SOLID. Thank you.

File: src/com/company/balance/GiftCardBalance.java (hunk 3 above)
Comment (mahir0837) [reply src/com/company/balance/GiftCardBalance.java]: done

**Review by ihccik — APPROVED** [review]
